Cabal/doc/developing-packages.rst

```diff
@@ -1,83 +1,112 @@
-   but can be customized if necessary.
+The ``cabal init`` command creates the necessary files for a Cabal
+package. It has an interactive mode which you can read more about below,
+but for now we're going to let it choose defaults for us.
 
-We can create both files manually or we can use ``cabal init`` to create
-them for us.
+::
 
-Using "cabal init"
-------------------
+    $ cabal init --non-interactive
+    # You can also use -n which is the short version of --non-interactive
 
-The ``cabal init`` command is interactive. It asks us a number of
-questions starting with the package name and version.
+This will create the ``Main.hs`` Haskell file and a few others that are
+part of all Cabal packages. By default ``cabal init`` will create an
+executable in the top level project directory.
 
 ::
 
-    $ cabal init
-    Package name [default "proglet"]?
-    Package version [default "0.1"]?
-    ...
+    Generating Setup.hs...
+    Generating CHANGELOG.md...
+    Generating Main.hs...
+    Generating proglet.cabal...
 
-It also asks questions about various other bits of package metadata. For
-a package that you never intend to distribute to others, these fields
-can be left blank.
+Check out the files created using the ``ls`` command.
 
-One of the important questions is whether the package contains a library
-or an executable. Libraries are collections of Haskell modules that can
-be re-used by other Haskell libraries and programs, while executables
-are standalone programs.
+
+Running the program
+-------------------
+
+Now that we have our Haskell code and the extra files that Cabal needs we
+can build and run our project. Since we have an executable we can use
+``cabal run`` which will build our executable (and re-build it if we've
+made any changes) and then run the binary.
 
 ::
 
-    What does the package build:
-       1) Library
-       2) Executable
-    Your choice?
+   $ cabal run
+   <Stuff about resolving dependencies, build profile, compliling, linking>
+   Hello, Haskell!
 
-For the moment these are the only choices. For more complex packages
-(e.g. a library and multiple executables or test suites) the ``.cabal``
-file can be edited afterwards.
 
-Finally, ``cabal init`` creates the initial ``proglet.cabal`` and
-``Setup.hs`` files, and depending on your choice of license, a
-``LICENSE`` file as well.
+About the Cabal package structure
+---------------------------------
+
+It is assumed that all the files that make up a package live under a common
+project root directory (apart from external dependencies). This simple
+example will have all the project files in one directory, but most packages
+will use one or more subdirectories.
+
+Cabal needs two extra files in the project's root directory:
+
+-  ``proglet.cabal``: contains package metadata and build information.
+
+-  ``Setup.hs``: contains a few standardized lines of code, but can be
+   customized if necessary.
+
+
+Running ``cabal init`` interactively
```

Comment:
I guess section above sections can go after this one because they are connected to `interactive` mode as well. For me this looks better structured:
* Using `cabal init` non-interactively (if this is the default one)
* Using `cabal init` interactively
* Building and running 
* ...

What do you think?

Reply:
Much better, reworded the sections a bit as well. PTAL!